src/lib.rs

```diff
@@ -1,6 +1,560 @@
+                new_string.replace_range(
+                    (i.index as usize)..( (i.index as usize)+1 as usize),
+                &y.to_string())
+            },
+            _ => {
+                panic!("Unrecoginzed Opperation")
+            }
+        }
+    }
+    return new_string.clone();
+}
+
+pub trait StringDiffAlgorithm {
+    //fn diff<'a>(s1: &'a str, s2: &'a str) -> Vec<StringDiffOp>;
+    fn diff<'a>(&self, s1: &'a str, s2: &'a str) -> Vec<StringDiffOp>;
+    fn distance<'a>(&self, s1: &'a str, s2: &'a str) -> usize;
+}
+
+pub struct HammingDistance {}
+impl HammingDistance{}
+impl StringDiffAlgorithm for HammingDistance {
+    fn diff<'a>(&self, s1: &'a str, s2: &'a str) -> Vec<StringDiffOp>{
+        let mut opp_vec : Vec<StringDiffOp> = Vec::new();
+        if s1.len() != s2.len(){
+            panic!("Strings must be same length");
+        }
+        else{
+            for i in 0..s1.len(){
+                if s1.chars().nth(i).unwrap() != s2.chars().nth(i).unwrap(){
+                    let new_opp = StringDiffOp{
+                        kind : StringDiffOpKind::Substitute(
+                            s1.chars().nth(i).unwrap(),
+                            s2.chars().nth(i).unwrap() ),
+                        index : i as isize
+                    };
+                    opp_vec.push(new_opp)
+                }
+            }
+        }
+        return opp_vec;
+    }        
+
+    
+    fn distance<'a>(&self, s1: &'a str, s2: &'a str) -> usize{
+        let mut edit_distance = 0;
+        if s1.len() != s2.len(){
+            panic!("Strings must be same length");
+        }
+        else{
+            for i in 0..s1.len(){
+                if s1.chars().nth(i).unwrap() != s2.chars().nth(i).unwrap(){
+                    edit_distance += 1;
+                }
+            }
+        }
+        return edit_distance;
+    }
+}
+
+
+pub struct LevenshteinDistance {}
+impl LevenshteinDistance{
+    pub(crate) fn min_dist(x: usize, y: usize, z: usize) -> usize {
+        if x <= y && x <= z{
+            return x;
+        }
+        else if y <= x && y <= z {
+            return y;
+        }
+        else{
+            return z;
+        }
+    }
+
+    pub(crate) fn min_dist_with_dir(x: usize, y: usize, z: usize) -> (usize,char){
+        if x <= y && x <= z{
+            return (x, '^');
+        }
+        if y <= x && y <= z {
+            return (y, '<');
+        }
+        if z <= x && z <= y {
+            return (z, '\\');
+        }
+        return (0, 'x')
+    }
+
+    pub(crate) fn print_vector<T: std::fmt::Debug>(my_vector: &[T]){
+        for i in my_vector.iter(){
+            println!("{:?}", i);
+        }
+    }
+
+    pub(crate) fn get_operations(my_opp: &Vec<Vec<char>>, 
+                                    left_string: &str,
+                                    top_string: &str) -> Vec<StringDiffOp> {
+                        
+        let mut my_opp_vector : Vec<StringDiffOp> = Vec::new();
```

Comment:
I think this would make more sense to be called `diff_ops` instead (operations also only has 1 p)